src/reader.rs

```diff
@@ -170,6 +172,20 @@ pub fn integer_parser(input: &str) -> IResult<&str, i32> {
     );
     integer_lexer(input).map(|(rest, digits)| (rest, digits.parse().unwrap()))
 }
+
```

Comment:
Hey there, you don't actually need a boolean parser -- you can basically directly make a reader similar to, say, nil reader, like this:

```rust
pub fn try_read_boolean(input: &str) -> IResult<&str, Value> {
    named!(true<&str, &str>, tag!("true"));
    named!(false<&str,&str>, tag!("false"));
    named!(bool_parser<&str,&str>, alt!(true | false))
    
    let (rest_input,bool) = bool_parser(input)?;
    
    Ok((rest_input,bool.parse().unwrap()))
}
```